synapse/handlers/message.py

```diff
@@ -130,6 +124,8 @@ async def get_room_data(
                 [membership_event_id], StateFilter.from_types([key])
             )
             data = room_state[membership_event_id].get(key)
+        else:
+            raise SynapseError(403, "User not in room", errcode=Codes.FORBIDDEN)
```

Comment:
This looked like a behavior change to me, but I think it is just ensuring that we raise a `SynapseError` instead of a `NameError` (from `data` not being defined).

Reply:
Actually, I've just realised that `check_user_in_room` will either return a `Membership.JOIN` or `Membership.LEAVE`, or raise an `AuthError` otherwise. `check_user_in_room_or_world_readable`, which is the calling function, will catch that `AuthError`, and then either raise another `AuthError`, or return `Membership.JOIN`.

Therefore, there isn't a case where `data` will be something other than `Membership.JOIN` or `Membership.LEAVE`. Either it's one of those are an `AuthError` is raised with essentially the same error message I've already put in.

However, it's still probably a good idea to cover our bases here in case the code below changes. Or, we could just add a comment explaining why there isn't an `else` statement, but my IDE would still complain :)

Reply:
Having the `else` statement seems fine to me, although I suppose it could hide an error in those methods.

Reply:
True. I can stick a log line in?

Reply:
Have added, let me know what you think.

Reply:
@anoadragon453 I think those changes seem OK.